flink-connector-bigquery/src/test/java/com/google/cloud/flink/bigquery/sink/serializer/BigQuerySchemaProviderTest.java

```diff
@@ -0,0 +1,502 @@
+/*
+ * Copyright (C) 2023 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.flink.bigquery.sink.serializer;
+
+import com.google.protobuf.DescriptorProtos.FieldDescriptorProto;
+import com.google.protobuf.Descriptors.Descriptor;
+import com.google.protobuf.Descriptors.FieldDescriptor;
+import org.apache.avro.Schema;
+import org.junit.Test;
+
+import static com.google.cloud.flink.bigquery.sink.serializer.AvroToProtoSerializerTestSchemas.getAvroSchemaFromFieldString;
+import static com.google.common.truth.Truth.assertThat;
+import static org.junit.Assert.assertThrows;
+
+/** Tests for {@link BigQuerySchemaProvider}. */
+public class BigQuerySchemaProviderTest {
+
+    // ------------------ Test Primitive Data Types (Nullable and Required) ------------------
+    @Test
+    public void testPrimitiveTypesConversion() {
+        Descriptor descriptor =
+                AvroToProtoSerializerTestSchemas.getSchemaWithRequiredPrimitiveTypes()
+                        .getDescriptor();
+        assertPrimitive(descriptor, false);
+    }
+
+    @Test
+    public void testRemainingPrimitiveSchemaConversion() {
+        Descriptor descriptor =
+                AvroToProtoSerializerTestSchemas.getSchemaWithRemainingPrimitiveTypes()
+                        .getDescriptor();
+        assertRemainingPrimitive(descriptor, false);
+    }
+
+    @Test
+    public void testNullablePrimitiveTypesConversion() {
+        Descriptor descriptor =
+                AvroToProtoSerializerTestSchemas.getSchemaWithNullablePrimitiveTypes()
+                        .getDescriptor();
+        assertPrimitive(descriptor, true);
+    }
+
+    @Test
+    public void testUnionOfRemainingPrimitiveSchemaConversion() {
+        Descriptor descriptor =
+                AvroToProtoSerializerTestSchemas.getSchemaWithUnionOfRemainingPrimitiveTypes()
+                        .getDescriptor();
+        assertRemainingPrimitive(descriptor, true);
+    }
+
+    private void assertPrimitive(Descriptor descriptor, Boolean isNullable) {
```

Comment:
All test methods at top, all supporting methods at bottom. Applicable to this entire file.

Reply:
Fixed.